CHANGELOG.md

```diff
@@ -1,926 +1,13 @@
-- #97 Import custom Sass variable overrides before default variables are defined @montchr and @ptvandi
-
-### Additions
-- #117 Add links to docs for setting up GH pages locally @gnarea
-- #95 Add SEO and 'lang' param for `_config` @gebeto
-
-## v0.2.3
-
-### Enhancements
-- Adds ability to use Google Analytics tracking by @pmarsceill
-
-### Bug fixes
-- Fixes 404 error for "/assets/js//search-data.json" by @stephenedmondson
-- Fixes #80 Single quotes in the string were unescaped and ruby attempted variable substitution of amp within it (which failed) by @novelistparty
-- Fixes bug that would only show 2 or more search results (not one) by @ilivewithian
-- Fixes a typo on the layout example by @woernfl
-- Fixes #78 Page scroll position too far down on load by @pmarsceill
-- Fixds ability to nest ul in ol without breaking style or counters
-
-### Dependency updates
-- Bumps stylelint dependency from 9.9.0 to 9.10.1
-
-## v0.2.2
-
-- Bumps stylelint-config-primer to 3.0.1 #44
-- Bumps bundler req to 2.0.1 #61
-- Adds custom 404 page
-- Excludes package-lock.json from jekyll build #47
-- Fixes keyboard scrolling / focus #48
-- Adds ARIA roles to navigation elements
-- Adds support for optional page description metadata (if present in yaml front matter)
-- Addresses some issues with search in #46
-- Option to hide TOC on parent pages if turned off in page's YAML front matter #30
-- Option to suppress an item from being indexed by search if present in page's YAML front matter #32
-
-## v0.2.1
-
-This update fixes security vulnerablities in the lodash sub-dependency and bumps other dev dependencies to their latest version.
-
-## v0.2.0
-
-Adds:
-- Dark mode via `color_scheme` parameter
-- Ability to exclude a page from the main nav with `nav_exclude` parameter closes #21
-- Ability for create children of children pages (3 nav levels) closes #25
-
-Changes:
-- Permalink structure for tiered navigation has been updated
-- Some colors have been updated for consistency / accessibility
-
-## v0.1.6
-
-### Added
-
-- Support for task list styles #19
-- Configuration docs
-- Configuration option to enable / disable search
-- Normalize.scss dependency pulled into project #16 #
-
-### Fixed
-
-- Layout bug in navigation #17
-
-## v0.1.5
-
-Major changes:
-
-- Fixed bug where the rake task would fail when the assets/js directory didn't exist
-
-## v0.1.4
-
-Major changes:
-- Adds Rake as a runtime dependency
-- Definition list styled
-- Sidebar and support cleaned up for smaller screen support
-- Updated some stale docs
-
-## v0.1.3
-
-Major changes:
-- Fix path problems, typos, and general clean-up for OSS.
-
-## v0.1.2
-
-Fix paths when deployed to gh-pages
-
-## v0.1.1
-
-Major updates:
-- Adds search to mobile nav
-- Pulls footer to bottom of the page on mobile (not hidden in nav)
-
-Minor updates:
-- Cleans up h1 typography spacing
+{% assign logs = site.changelog | reverse %}
+{% for log in logs %}
+{{ log }}
+{% endfor %}
```

Comment:
I'd counter on two points: one, if we're tolerating frontmatter (which we must), I feel that the line between "content" and "template" blurs in a jekyll theme and don't personally agree that there's a strong need to separate them, given the purpose of this repository. My second argument, which I'd argue is the more salient one, is that separating it out the way you're suggesting feels more complicated and would increase friction in using this theme. Anything which requires introducing a separate build process, I'd recommend we hold off on until we are certain we'd need it.

@joeleenk thoughts?

Reply:
I guess my question is whether we prefer to:

1. pull in smaller parts, like an archive page with posts, which requires a template of some kind, or
2. have larger, complete markdown files that are fully human readable

I personally prefer smaller, well organized bits of content most of the time, but I'm not sure if any of our projects will have so much content that it matters. 

I guess we could keep one example of how to use the templates, and switch the changelog to be static.

Reply:
> My second argument, which I'd argue is the more salient one, is that separating it out the way you're suggesting feels more complicated and would increase friction in using this theme.

I agree this is not optimal, which is why it was my last suggestion. I'd prefer the first option I suggested--one single human-readable `CHANGELOG.md`. At least personally, I don't feel this makes the theme meaningfully more difficult to use or update--if anything it makes it easier because you only need to edit a single file vs create a new file, make sure it has appropriate naming, frontmatter, etc.

> if we're tolerating frontmatter (which we must), I feel that the line between "content" and "template" blurs in a jekyll theme

I'm not sure I understand the argument here. Frontmatter isn't "meant for human consumption," sure, but it's metadata that describes the content that follows it: While it might not be easy for a human to read, it does _convey information_, which IMO makes it meaningfully different from liquid markup, which conveys no meaningful or useful information to a human. A markdown file containing only liquid markup is not helpful to a person looking for documentation in the repo; it's an impediment.

> I personally prefer smaller, well organized bits of content most of the time

Conceptually I do too--it just feels nicer--but in this case I feel like it complicates the system for updating the docs which is going to make updates less likely, and I'm not sure what additional benefit it provides beyond making sense organizationally. Without a larger system (definitely beyond the scope of this) that handles the UX/UI of creating and managing these files, I'd rather structure docs in the simplest (i.e. fewest files) way possible.

IMO one of the primary goals for documentation is that it should be as easy to create, update, and access as we can possibly make it, which is why I'm skeptical of splitting these files up and using `.md` files as templates. It is nice to have a site with the docs, but it's _also_ very valuable to have easy-to-read docs available right in the repo (either cloned to your local or on github). What is the benefit to the maintainer and the consumer in making it more difficult/time consuming to get to the actual changelog?

Reply:
> What is the benefit to the maintainer and the consumer in making it more difficult/time consuming to get to the actual changelog?

That's fair. I think I misunderstood the discussion due to a too-hasty review; I think trying to be less 'clever' in the theme, in favor of having the basic process of "take these markdown files and make them into a nice site, with as minimal an install process as possible," is in line with the original goals we had for the initiative.